v2/bigtable-cdc-to-hbase/src/main/java/com/google/cloud/teleport/v2/templates/BigtableCdcToHbase.java

```diff
@@ -0,0 +1,257 @@
+ * change stream, converts change stream mutations to their nearest Hbase counterparts, and writes
+ * the resulting Hbase row mutations to Hbase.
+ *
+ * <p>In Bigtable, all writes to a single row on a single cluster will be streamed in order by
+ * commitTimestamp. This is only the case for instances with a single cluster, or if all writes to a
+ * given row happen on only one cluster in a replicated instance. This order is maintained in the
+ * Dataflow connector via key-ordered delivery, as the Dataflow connector emits ChangeStreamMutation
+ * records with the row key as the record key.
+ */
+@Template(
+    name = "bigtable-cdc-to-hbase",
+    category = TemplateCategory.STREAMING,
+    displayName = "Bigtable CDC to HBase Replicator",
+    description = "A streaming pipeline that replicates Bigtable change stream data to HBase",
+    optionsClass = BigtableCdcToHbase.BigtableToHbasePipelineOptions.class,
+    flexContainerName = "bigtable-cdc-to-hbase",
+    contactInformation = "https://cloud.google.com/support")
+public class BigtableCdcToHbase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BigtableCdcToHbase.class);
+  private static final String USE_RUNNER_V2_EXPERIMENT = "use_runner_v2";
+
+  /** Options to run pipeline with. */
+  // TODO: merge with BigtableCommon options after CDC GA
+  public interface BigtableToHbasePipelineOptions
+      extends DataflowPipelineOptions, ExperimentalOptions {
+    /** Bigtable change stream configs. */
+    @TemplateParameter.Text(
+        description = "Bigtable project id",
+        helpText = "GCP project id that contains Bigtable instance")
+    String getBigtableProjectId();
+
+    void setBigtableProjectId(String bigtableProjectId);
+
+    @TemplateParameter.Text(
+        description = "Bigtable instance",
+        helpText = "Bigtable instance to stream changes from")
+    String getInstanceId();
+
+    void setInstanceId(String instanceId);
+
+    @TemplateParameter.Text(
+        description = "Table id",
+        helpText = "The name of the Bigtable table whose changes to stream")
+    String getTableId();
+
+    void setTableId(String tableId);
+
+    @TemplateParameter.Text(
+        optional = true,
+        description = "App profile id",
+        helpText =
+            "Bigtable's app profile id needs to have single-cluster routing with single-row transactions allowed")
+    @Default.String("default")
+    String getAppProfileId();
+
+    void setAppProfileId(String appProfileId);
+
+    /** Hbase specific configs. Mirrors configurations on hbase-site.xml. */
+    @TemplateParameter.Text(
+        description = "Zookeeper quorum host",
+        helpText = "Zookeeper quorum host, corresponds to hbase.zookeeper.quorum host")
+    String getHbaseZookeeperQuorumHost();
+
+    void setHbaseZookeeperQuorumHost(String hbaseZookeeperQuorumHost);
+
+    @TemplateParameter.Text(
+        description = "Zookeeper quorum port",
```

Comment:
Should this be `optional`? It has a default value for the port, so no need to set.